src/coreclr/jit/importer.cpp

```diff
@@ -18417,6 +18407,38 @@ void Compiler::impWalkSpillCliqueFromPred(BasicBlock* block, SpillCliqueWalker*
     assert(impSpillCliqueGetMember(SpillCliquePred, block) != 0);
 }
 
+//------------------------------------------------------------------------
+// impStringLiteralLength: create GenTreeIntCon node for the given string
+//    literal to store its length.
+//
+// Arguments:
+//    node  - string literal node.
+//
+// Return Value:
+//    GenTreeIntCon node with string's length as a value or null.
+//
+GenTreeIntCon* Compiler::impStringLiteralLength(GenTreeStrCon* node)
+{
+    int             length = -1;
+    const char16_t* str    = info.compCompHnd->getStringLiteral(node->gtScpHnd, node->gtSconCPX, &length);
+    if (length >= 0)
+    {
+        GenTreeIntCon* iconNode = gtNewIconNode(length);
+
+        // str can be NULL for dynamic context
+        if (str != nullptr)
+        {
+            JITDUMP("Optimizing '\"%ws\".Length' to just '%d'\n", str, length)
```

Comment:
NIT: this doesn't actually do any optimization, so maybe change this to
```suggestion
            JITDUMP("String '\"%ws\".Length' is '%d'\n", str, length)
```